File: mobile/src/services/backend/VoiceAgentClient.ts
import { VoiceSummaryApiResponse, VoiceSummaryRequest } from "./contracts";
import { BackendClient } from "./BackendClient";

export class VoiceAgentClient {
  constructor(private readonly backend = new BackendClient()) {}

  summarize(request: VoiceSummaryRequest): Promise<VoiceSummaryApiResponse> {
    return this.backend.post<VoiceSummaryApiResponse>("/voice/summary", request);
  }
}

File: ACM2/Services/Backend/VoiceAgentClient.swift
import Foundation

enum VoiceAgentContext: String, Codable {
    case fuel
    case maintenance
    case copilot
}

private struct VoiceSummaryRequest: Codable {
    let userId: String
    let context: VoiceAgentContext
    let transcript: String
}

struct VoiceAgentClient {
    private let backend: BackendClient

    init(backend: BackendClient = BackendClient()) {
        self.backend = backend
    }

    func summarize(
        userId: String,
        context: VoiceAgentContext,
        transcript: String
    ) async throws -> VoiceSummaryResponse {
        try await backend.post(
            "/voice/summary",
            body: VoiceSummaryRequest(userId: userId, context: context, transcript: transcript)
        )
    }
}
